rep_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The go-zeromq Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package zmq4_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-zeromq/zmq4"
+)
+
+func TestIssue99(t *testing.T) {
+	var (
+		wg     sync.WaitGroup
+		outMsg zmq4.Msg
+		inMsg  zmq4.Msg
+	)
+
+	ep, err := EndPoint("tcp")
+	if err != nil {
+		t.Errorf("could not find endpoint: %v", err)
+	}
+
+	requester := func() {
+
+		req := zmq4.NewReq(context.Background())
+		defer req.Close()
+		defer wg.Done()
+
+		err := req.Dial(ep)
+		if err != nil {
+			t.Errorf("could not dial: %v", err)
+		}
+
+		// Test message w/ 3 frames
+		outMsg = zmq4.NewMsgFromString([]string{"ZERO", "Hello!", "World!"})
+		err = req.Send(outMsg)
+		if err != nil {
+			t.Errorf("failed to send: %v", err)
+		}
+
+		inMsg, err = req.Recv()
+		if err != nil {
+			t.Errorf("failed to recv: %v", err)
+		}
+	}
+
+	responder := func() {
+
+		rep := zmq4.NewRep(context.Background())
+		defer rep.Close()
+		defer wg.Done()
+
+		err := rep.Listen(ep)
+		if err != nil {
+			t.Errorf("could not dial: %v", err)
+		}
+
+		//  Wait for next request from client
+		msg, err := rep.Recv()
+		if err != nil {
+			t.Errorf("could not recv request: %v", err)
+		}
+
+		//  Send reply back to client
+		err = rep.Send(msg)
+		if err != nil {
+			t.Errorf("could not send reply: %v", err)
+		}
+		time.Sleep(2 * time.Millisecond)
```

Comment:
perhaps I'm missing something (wouldn't be the first), but it seems to me this `sleep` isn't necessary, thanks to the `WaitGroup`, is it?

Reply:
True.